tests/kafkatest/services/connect.py

```diff
@@ -145,7 +158,7 @@ def restart_node(self, node, clean_shutdown=True):
     def clean_node(self, node):
         node.account.kill_process("connect", clean_shutdown=False, allow_fail=True)
         self.security_config.clean_node(node)
-        all_files = " ".join([self.CONFIG_FILE, self.LOG4J_CONFIG_FILE, self.PID_FILE, self.LOG_FILE, self.STDOUT_FILE, self.STDERR_FILE] + self.config_filenames() + self.files)
+        all_files = " ".join([self.CONFIG_FILE, self.LOG4J_CONFIG_FILE, self.PID_FILE, self.LOG_FILE, self.STDOUT_FILE, self.STDERR_FILE, self.EXTERNAL_FILE] + self.config_filenames() + self.files)
```

Comment:
should we delete the EXTERNAL file only if it was created by the `self.external_config_template_func` flag? I think the rm command will fail if the file doesn't exist.

Reply:
At least on MacOS, `rm -rf x` does not fail if the file doesn't exist, but it does fail if it exists but can't be deleted.

Reply:
I also checked the debug build output of multiple tests, including those that did and did not generate this file. No problems removing the file.

Reply:
makes sense. Thanks, @rhauch!